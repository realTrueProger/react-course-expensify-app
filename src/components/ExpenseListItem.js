import React from 'react';
import {Link} from 'react-router-dom';


const ExpensesListItem = ( { description, amount, createdAt , id}) => (
    <div>
        <h2>Description: {description}</h2>
        <Link to={`/edit/${id}`}>edit</Link>
        <p>Amount: {amount} CreatedAt: {createdAt}</p>
    </div>
);


export default ExpensesListItem;