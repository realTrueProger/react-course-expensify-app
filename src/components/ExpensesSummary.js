import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import total from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        {props.expenses.length > 0 &&
            <p>You have {props.expenses.length} expense(s) with total cost ${total(props.expenses)}</p>
        }
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filter)
});

export default connect(mapStateToProps)(ExpensesSummary);