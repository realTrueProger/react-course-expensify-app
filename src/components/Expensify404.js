import React from 'react';
import { Link} from 'react-router-dom';

const Expensify404 = () => (
    <div>
        404 not found - <Link to="/">Go home</Link>
    </div>
);

export default Expensify404;