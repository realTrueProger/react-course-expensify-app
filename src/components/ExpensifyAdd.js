import React from 'react';
import {connect} from 'react-redux';
import {addExpense} from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const ExpensifyAdd = (props) => (
    <div>
        <h1>Add expense</h1>
        <ExpenseForm 
            onSubmit={(expense) => {
                props.dispatch(addExpense(expense));
                props.history.push('/');
            }}
        />
    </div>
);

export default connect()(ExpensifyAdd);