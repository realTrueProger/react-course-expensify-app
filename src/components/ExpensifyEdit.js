import React from 'react';
import { connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense} from '../actions/expenses';

const ExpensifyEdit = (props) => (
    <div>
        <ExpenseForm 
            expense={props.expense}
            onSubmit={(expense) => {
                props.dispatch(editExpense(props.expense.id, expense));
                props.history.push('/');
            }}
        />
        <button onClick={() => {
            props.dispatch(removeExpense( {id : props.expense.id}));
            props.history.push('/');
        }}>remove</button>
    </div>
);

const mapStateTOProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});



export default connect(mapStateTOProps)(ExpensifyEdit);