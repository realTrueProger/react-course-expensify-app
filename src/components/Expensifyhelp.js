import React from 'react';

const ExpensifyHelp = () => (
    <div>
        help
    </div>
);

export default ExpensifyHelp;