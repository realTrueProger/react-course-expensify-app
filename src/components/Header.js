import React from 'react';
import { NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify app</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home page</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
    </header>
);

export default Header;