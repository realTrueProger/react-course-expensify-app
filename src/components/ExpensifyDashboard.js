import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpensifyDashboard = () => (
    <div>
        <h1>Expenses dashboard</h1>
        <ExpenseListFilters />
        <ExpensesSummary />
        <ExpenseList />
    </div>
);

export default ExpensifyDashboard;