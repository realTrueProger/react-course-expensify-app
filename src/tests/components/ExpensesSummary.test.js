import React from 'react';
import {shallow} from 'enzyme';
import { ExpensesSummary} from '../../components/ExpensesSummary';
import fixtures from '../fixtures/expenses';

test('should render 3 expenses total with $21000' , () => {
    const wrapper = shallow(<ExpensesSummary expenses={fixtures} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render 1 expense total with $10000' , () => {
    const wrapper = shallow(<ExpensesSummary expenses={[fixtures[0]]} />);
    expect(wrapper).toMatchSnapshot();
});