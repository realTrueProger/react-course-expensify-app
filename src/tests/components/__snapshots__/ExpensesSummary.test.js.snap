// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense total with $10000 1`] = `
<div>
  <p>
    You have 
    1
     expense(s) with total cost $
    10000
  </p>
</div>
`;

exports[`should render 3 expenses total with $21000 1`] = `
<div>
  <p>
    You have 
    3
     expense(s) with total cost $
    21000
  </p>
</div>
`;
